src/twister2/platform_specification.py

```diff
@@ -113,12 +116,46 @@ class PlatformSchema(Schema):
 
 def discover_platforms(directory: Path) -> Generator[PlatformSpecification, None, None]:
     """Return platforms from given directory"""
-    for file in directory.glob('*/*/*.yaml'):
+    for spec_file in directory.glob('*/*/*.yaml'):
         try:
-            yield PlatformSpecification.load_from_yaml(str(file))
+            platform = PlatformSpecification.load_from_yaml(str(spec_file))
         except Exception as e:
             logger.exception('Cannot read platform definition from yaml: %s', e)
             raise
+        yield platform
+        if '@' not in platform.identifier:
+            yield from discover_platform_revisions(platform, spec_file.parent)
+
+
+def discover_platform_revisions(
+    platform: PlatformSpecification, directory: Path
+) -> Generator[PlatformSpecification, None, None]:
+    """
+    Return all revisions for platform specification in the directory.
+
+    :param platform: platform specification
+    :param directory: directory to search revision in
+    :return: all revisions for given platform
+    """
+    revision_patterns: list[str] = [
+        '[A-Z]',
+        '(0|[1-9][0-9]*)(_[0-9]+)*(_[0-9]+)*'
```

Comment:
`(_[0-9]+)*` and `(_[0-9]+)*(_[0-9]+)*` give the same output

Reply:
I simplify the pattern and the code. It matches version like `0` and `012`.

Reply:
ok, I see it now